backend/PythonClient/multirotor/airsim_application.py

```diff
@@ -26,10 +29,15 @@ def __init__(self):
         self.graph_dir = os.path.join("Debug")
         self.current_time_string = ""
         self.cesium_origin = self.get_cesium_origin()
-        report_root_path = os.path.join(os.path.expanduser("~"), "Documents", "AirSim", "report")
-        if not os.path.exists(report_root_path):
-            os.mkdir(report_root_path)
-        self.dir_path = report_root_path
+        # report_root_path = os.path.join(os.path.expanduser("~"), "Documents", "AirSim", "report")
+        # if not os.path.exists(report_root_path):
+        #     os.mkdir(report_root_path)
+        # self.dir_path = report_root_path
+
+        # MongoDB setup
+        self.mongo_client = MongoClient("your_mongo_connection_string") # replace MongoDB connection string
+        self.db = self.mongo_client["your_database_name"] # replace with database name
+        self.collection = self.db["your_collection_name"] # replace with collection name
```

Comment:
Instead of having your airsim_application.py depend directly on mongodb or (as in commented out code) on saving to local disk, you should abstract out the 'saving' operation. Specifically, you can create an abstraction (abstract class or a python 'protocol' https://typing.readthedocs.io/en/latest/spec/protocol.html) defining the 'save' operation. You can then have multiple implementations of this protocol, and use the one appropriate for the situation. So you would have something like:
- PythonClient/data_storage
     - data_storage.py (define your protocol here, as a class)
     - mongodb_storage.py (your mongodb code, wrapped in a class, here)
     - file_storage.py (the commented out code, wrapped in a class, here)
When instantiating AirSimApplication, provide an object to be used for storage (either mongodb_storage or file_storage), as a data_storage abstraction. This way, AirSimApplication will only depend on the abstraction and not a specific choice of where /how to store the data.